### ISSUE TITLE
crash on non-positive duration

### ISSUE THREAD

**Issue by RubenPX** [zkxs/vrcx-optimal-time#15]
```
thread 'main' panicked at 'got a non-positive duration for randomUser', src\main.rs:129:33
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by zkxs** [comment]
Hmm, looks like the me of 3 months ago wrote a [sanity-check which is failing](https://github.com/zkxs/vrcx-optimal-time/blob/0.3.1/src/main.rs#L129). I'll look into it in more detail later.

**Comment by zkxs** [comment]
I've changed the sanity-check logic a bit to hopefully be a little more tolerant of what I *think* is going wrong here, and the change is in [the new release](https://github.com/zkxs/vrcx-optimal-time/releases/tag/0.4.0).

**Comment by RubenPX** [comment]
Nope, still crashing

```
thread 'main' panicked at 'Got a negative (-17740554ms) duration for mia25za. This should not happen as long as events are indexed in the table in chronological order.', src\main.rs:135:33
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

if you need more details or you need make any sql query, you can paste here a query

Extra, if you need see vrcx db tables, you can contact me:
- Discord: [RubenPX#1568](http://discordapp.com/users/RubenPX#1568)
- Telegram: @KryptoPX

**Comment by RubenPX** [comment]
Nope, still crashing

```
thread 'main' panicked at 'Got a negative (-17740554ms) duration for hiddenName. This should not happen as long as events are indexed in the table in chronological order.', src\main.rs:135:33
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

if you need more details or you need make any sql query, you can paste here a query

Extra, if you need see vrcx db tables, you can contact me:
- Discord: [RubenPX#1568](http://discordapp.com/users/RubenPX#1568)
- Telegram: @KryptoPX

**Comment by zkxs** [comment]
I've got one last idea before I'm completely clueless, after which I'd definitely need to get this reproduced locally to debug.

I've implemented my idea in [release 0.4.1](https://github.com/zkxs/vrcx-optimal-time/releases/tag/0.4.1). Basically, if I'm relying on the rows being sorted then [why the hell wasn't I sorting them](https://github.com/zkxs/vrcx-optimal-time/commit/e1b23e1eb9097c018b844dbf920b3045c1396671#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fc)?
